cpu/cc2538/radio/cc2538_rf_radio_ops.c

```diff
@@ -0,0 +1,471 @@
+                RFCORE_SFR_RFST = ISTXON;
+            }
+            else {
+                cc2538_rf_dev.cb(&cc2538_rf_dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
+            }
+        }
+        else {
+            cc2538_cca_status = BOOLEAN(RFCORE->XREG_FSMSTAT1bits.CCA) && RFCORE->XREG_RSSISTATbits.RSSI_VALID;
+            cc2538_rf_dev.cb(&cc2538_rf_dev, IEEE802154_RADIO_CONFIRM_CCA);
+        }
+    }
+}
+
+static int _off(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    return -ENOTSUP;
+}
+
+static bool _get_cap(ieee802154_dev_t *dev, ieee802154_rf_caps_t cap)
+{
+    (void) dev;
+    switch(cap) {
+        case IEEE802154_CAP_24_GHZ:
+        case IEEE802154_CAP_IRQ_TX_DONE:
+        case IEEE802154_CAP_IRQ_CCA_DONE:
+        case IEEE802154_CAP_AUTO_CSMA:
+            return true;
+        default:
+            return false;
+    }
+}
+
+static int _set_hw_addr_filter(ieee802154_dev_t *dev, network_uint16_t *short_addr,
+                               eui64_t *ext_addr, uint16_t pan_id)
+{
+    (void) dev;
+    if (short_addr) {
+        RFCORE_FFSM_SHORT_ADDR0 = short_addr->u8[1];
+        RFCORE_FFSM_SHORT_ADDR1 = short_addr->u8[0];
+    }
+
+    if (ext_addr) {
+        RFCORE_FFSM_EXT_ADDR0 = ext_addr->uint8[7];
+        RFCORE_FFSM_EXT_ADDR1 = ext_addr->uint8[6];
+        RFCORE_FFSM_EXT_ADDR2 = ext_addr->uint8[5];
+        RFCORE_FFSM_EXT_ADDR3 = ext_addr->uint8[4];
+        RFCORE_FFSM_EXT_ADDR4 = ext_addr->uint8[3];
+        RFCORE_FFSM_EXT_ADDR5 = ext_addr->uint8[2];
+        RFCORE_FFSM_EXT_ADDR6 = ext_addr->uint8[1];
+        RFCORE_FFSM_EXT_ADDR7 = ext_addr->uint8[0];
+    }
+
+    RFCORE_FFSM_PAN_ID0 = pan_id;
+    RFCORE_FFSM_PAN_ID1 = pan_id >> 8;
+    return 0;
+}
+
+static int _confirm_on(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    /* TODO */
+    return 0;
+}
+
+static int _request_on(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    /* TODO */
+    return 0;
+}
+
+static int _set_cca_mode(ieee802154_dev_t *dev, ieee802154_cca_mode_t mode)
+{
+    (void) dev;
+    uint8_t tmp = 0;
+    switch (mode) {
+        case IEEE802154_CCA_MODE_ED_THRESHOLD:
+            tmp = 1;
+            break;
+        case IEEE802154_CCA_MODE_CARRIER_SENSING:
+            tmp = 2;
+            break;
+        case IEEE802154_CCA_MODE_ED_THRESH_AND_CS:
+            tmp = 3;
+            break;
+        case IEEE802154_CCA_MODE_ED_THRESH_OR_CS:
+            return -ENOTSUP;
+    }
+
+    RFCORE_XREG_CCACTRL1 &= CC2538_CCA_MODE_MASK;
+    RFCORE_XREG_CCACTRL1 |= (tmp << CC2538_CCA_MODE_POS);
+
+    return 0;
+}
+
+static int _set_rx_mode(ieee802154_dev_t *dev, ieee802154_rx_mode_t mode)
+{
+    (void) dev;
+
+    bool promisc = false;
+    bool ack_filter = true;
+    switch (mode) {
+        case IEEE802154_RX_AACK_DISABLED:
+            RFCORE->XREG_FRMCTRL0bits.AUTOACK = 0;
+            break;
+        case IEEE802154_RX_AACK_ENABLED:
+            RFCORE->XREG_FRMCTRL0bits.AUTOACK = 1;
+            RFCORE_XREG_FRMCTRL1 &= ~0x4;
```

Comment:
Magic numbers? Can a define be used here?

Reply:
here as well

Reply:
done!